### PR TITLE
Notify user resource

### PR DESCRIPTION
**Pull request by viktormuller**
First version of new Notify features and some improvements
- User resource
- Segment resource
- Autogenerated Endpoint ID
- Some other fixes and improvements

Expect more updates on Monday.

### CODE REVIEW

File: notifications/rest/bindings/create-binding/create-binding.3.x.js [hunk 1]
```diff
@@ -9,10 +9,10 @@ var client = new Twilio(accountSid, authToken);
 var service = client.notify.services('ISxxx');
 
 service.bindings.create({
-    endpoint: 'xxx',
+    endpoint: 'eyJjdHkiOiJ0d2lsaW8tZnBhO3Y9MSIsInR5cCI6IkpXVCIsImFsZyI6IkhTMjU2In0.eyJncmFudHMiOnsiY2hhdCI6eyJzZXJ2aWNlX3NpZCI6IklTNmIwZmE2MDgzOWM1MTFlNWExNTFmZWZmODE5Y2RjOWYiLCJlbmRwb2ludF9pZCI6InR3aTEtMTdjNTkyM2Y5NTBiNGJhYjllZjAwZTJlZjdkYTBkYTIifSwiaWRlbnRpdHkiOiJOb3RpZmljYXRpb25MaWJyYXJ5VW5pdFRlc3RlcjEifSwic3ViIjoiQUM3OGU4ZTY3ZmMwMjQ2NTIxNDkwZmI5OTA3ZmQwYzE2NSIsImlzcyI6IlNLMmQxYWRmOGIxYzY0YTYzM2U1ZjE4NjU5ZjdjZDE5MTEiLCJleHAiOjE0NzgyNTI4NTl9.1i256CwtacP859Xh-6cEZVio4oohxL-od9krzsa4DnY',
```
Comment (coderkat) [inline notifications/rest/bindings/create-binding/create-binding.3.x.js]: Is this endpoint unique to a service binding (will it be different per user/instance)? If so, can we make it read more like a placeholder, a la 'XXXXXXXXXXXXXXXXXXXXXXXX'

File: notifications/rest/bindings/create-binding/create-binding.3.x.js [hunk 2]
```diff
@@ -9,10 +9,10 @@ var client = new Twilio(accountSid, authToken);
...
     identity: "00000001",
-    bindingType: 'gcm',
-    address: 'xxx',
+    bindingType: 'apn',
+    address: '483161f7afe6f012aa5f172d3ca93f5c245e3daf34d3a911a41f41f6af3639f6',
```
Comment (coderkat) [inline notifications/rest/bindings/create-binding/create-binding.3.x.js]: Same as above: we generally prefer to keep unique IDs masked in our API snippets, unless they are the same for every use case

File: notifications/rest/bindings/list-binding/output/list-binding.json [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+    {
+      "endpoint": "eyJjdHkiOiJ0d2lsaW8tZnBhO3Y9MSIsInR5cCI6IkpXVCIsImFsZyI6IkhTMjU2In0.eyJncmFudHMiOnsiY2hhdCI6eyJzZXJ2aWNlX3NpZCI6IklTNmIwZmE2MDgzOWM1MTFlNWExNTFmZWZmODE5Y2RjOWYiLCJlbmRwb2ludF9pZCI6InR3aTEtMTdjNTkyM2Y5NTBiNGJhYjllZjAwZTJlZjdkYTBkYTIifSwiaWRlbnRpdHkiOiJOb3RpZmljYXRpb25MaWJyYXJ5VW5pdFRlc3RlcjEifSwic3ViIjoiQUM3OGU4ZTY3ZmMwMjQ2NTIxNDkwZmI5OTA3ZmQwYzE2NSIsImlzcyI6IlNLMmQxYWRmOGIxYzY0YTYzM2U1ZjE4NjU5ZjdjZDE5MTEiLCJleHAiOjE0NzgyNTI4NTl9.1i256CwtacP859Xh-6cEZVio4oohxL-od9krzsa4DnY",
       "notification_protocol_version": "default",
       "tags": [
         "new user",
         "preferred device"
       ],
       "date_updated": "2017-01-17T14:26:42Z",
-      "address": "+44XXXXXXXXXX",
+      "address": "+441234123456",
```
Comment (coderkat) [inline notifications/rest/bindings/list-binding/output/list-binding.json]: Again, it's better to not use a unique number in these snippets, unless every user's code needs exactly this address

File: notifications/rest/users/list-binding-of-user/output/list-binding.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      "identity": "0000001"
+    },
+        {
+      "endpoint": "jiewKjdHkiOiJ0d2lsaW8tZnBhO3Y9MSIsInR5cCI6IkpXVCIsImFsZyI6IkhTMjU2In0.eyJncmFudHMiOnsiY2hhdCI6eyJzZXJ2aWNlX3NpZCI6IklTNmIwZmE2MDgzOWM1MTFlNWExNTFmZWZmODE5Y2RjOWYiLCJlbmRwb2ludF9pZCI6InR3aTEtMTdjNTkyM2Y5NTBiNGJhYjllZjAwZTJlZjdkYTBkYTIifSwiaWRlbnRpdHkiOiJOb3RpZmljYXRpb25MaWJyYXJ5VW5pdFRlc3RlcjEifSwic3ViIjoiQUM3OGU4ZTY3ZmMwMjQ2NTIxNDkwZmI5OTA3ZmQwYzE2NSIsImlzcyI123NLMmQxYWRmOGIxYzY0YTYzM2U1ZjE4NjU5ZjdjZDE5MTEiLCJleHAiOjE0NzgyNTI4NTl9.1i256CwtacP859Xh-6cEZVio4oohxL-od9krz1231S1asd1",
+      "notification_protocol_version": "3",
+      "tags": [
+        "active_in_30d"
+      ],
+      "date_updated": "2017-01-16T14:26:42Z",
+      "address": "483161f7afe6f012aa5f172d3ca93f5c245e3daf34d3a911a41f41f6af3639f6",
```
Comment (coderkat) [inline notifications/rest/users/list-binding-of-user/output/list-binding.json]: Unique address should be masked?

**Comment by viktormuller** [comment]
Hi Kat,

I put in specific values as I think that helps people pattern match how a
certain value should look like and confirm that they have the right value
or not. These are not secrets so perfectly fine to have them (I even
changed them from their original value so there is even less of an issue).

Best regards

Viktor

**Comment by viktormuller** [comment]
@coderkat Added support for autogenerated endpoint IDs to the "Registering for iOS / Android push notifications" guides. 

I have not had a chance to verify that these changes compile especially the objective-C (the other two I took from tested Android and iOS quickstart apps). So if you can please, check that at least they compile.

**Comment by viktormuller** [comment]
Sorry forgot to update the server example in the previous commit. now it is there.

File: notifications/rest/bindings/create-binding/create-binding.6.x.py [hunk 5]
```diff
@@ -13,7 +13,9 @@
     identity="00000001",
     binding_type="apn",
     address="483161f7afe6f012aa5f172d3ca93f5c245e3daf34d3a911a41f41f6af3639f6",
-    tag="premium"
+    tag="preferred_device",
+    tag="new user"
+>>>>>>> deved/master
```
Comment (coderkat) [inline notifications/rest/bindings/create-binding/create-binding.6.x.py]: Please remove this merge line!

Comment (viktormuller) [inline notifications/rest/bindings/create-binding/create-binding.6.x.py]: oh damn, fixing, 1 min

Comment (viktormuller) [inline notifications/rest/bindings/create-binding/create-binding.6.x.py]: @coderkat Fixed, see below. thx

**Pull request merged**